src/rgw/motr/gc/gc.cc

```diff
@@ -43,51 +45,79 @@ void *MotrGC::GCWorker::entry() {
       uint32_t processed_count = 0;
       // form the index name
       iname = motr_gc->index_names[my_index];
-      ldpp_dout(dpp, 10) << __func__ << ": " << gc_thread_prefix
-        << worker_id << " Working on GC Queue: " << iname << dendl;
-      // time based while loop
-      do {
-        // fetch the next entry from index "iname"
-        // check if the object is ready for deletion
-        // for the motr object
-
-        // delete that object
-        // rc = dequeue();
-
-        // remove the entry from iname
+      ldpp_dout(dpp, 10) << gc_log_prefix
+                      << __func__ << ": Working on GC Queue: " << iname << dendl;
+      
+      bufferlist bl;
+      std::vector<std::string> keys(motr_gc->max_count + 1);
+      std::vector<bufferlist> vals(motr_gc->max_count + 1);
+      uint32_t rgw_gc_obj_min_wait = cct->_conf->rgw_gc_obj_min_wait;
+      keys[0] = obj_exp_time_prefix;
+      rc = motr_gc->store->next_query_by_name(iname,
+                                  keys, vals, obj_exp_time_prefix);
+      ldpp_dout(dpp, 0) << gc_log_prefix 
+                         << __func__ <<": next_query_by_name() rc=" << rc << dendl;
+      if (rc < 0) {
+        // In case of failure, worker will keep retrying till end_time
+        ldpp_dout(dpp, 0) << gc_log_prefix 
+                        << __func__ <<": ERROR: NEXT query failed. rc=" << rc << dendl;
+        continue;
+      }
 
-        // Exit the loop if required work is complete
+      // fetch entries as per defined in rgw_gc_max_trim_chunk from index iname
+      for (uint32_t j = 0; j < motr_gc->max_count &&
+                              !keys[j].empty() && keys[j] != obj_exp_time_prefix; j++) {
+        bufferlist::const_iterator blitr = vals[j].cbegin();
+        motr_gc_obj_info ginfo;
+        ginfo.decode(blitr);
+        // Check if the object is ready for deletion
+        if(ginfo.deletion_time + rgw_gc_obj_min_wait > std::time(nullptr)) {
```

Comment:
@jjxsg: Should this be > or <? I think, when object expiry time > current time, we should delete the object. And not otherwise. am I missing something?

Reply:
Its greater than `<` here just like you said.

Reply:
Let's assume
current time = t
deletion_time = t+10
rgw_gc_obj_min_wait = 20

So, we will delete object only if current time is greater than t+10+20 = t+30.

Reply:
ok